src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2936,6 +2935,13 @@ private static bool AttributeTypeNamesMatch(Type attributeType, Type reflectedAt
                     }
                 }
 
+                // Scoping the call to GetFields to a local function to limit the linker suppression
+#if !ES_BUILD_STANDALONE
+                [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:UnrecognizedReflectionPattern",
+                    Justification = "Nested type members will be preserved with the parent annotation")]
+#endif
```

Comment:
With the new linker this should not be necessary (both the suppression and local function)

Reply:
Is the new linker picked in the repo builds? I added cause the checks failed in the PR build (before you checked in the code though)

Reply:
Yes - it's auto updated - I think this is the commit which brought it into runtime... but I could be wrong: https://github.com/dotnet/runtime/commit/2f03850bade9930757281f8840b5c6745f3f9a3a